src/main/scala/io/iohk/ethereum/network/p2p/Message.scala

```diff
@@ -17,6 +17,8 @@ trait MessageSerializable extends Message {
 
   def toBytes: Array[Byte]
 
+  def underlyingMsg: Message
```

Comment:
SendMessage looks good, but only at the point where SendMessage is called, the implicit must be available. 
`import messages._`
`peer.ref ! SendMessage(GetBlockHeaders(...).)`

In this way the implicit happens outside network?